mmv1/third_party/terraform/provider/provider.go.tmpl

```diff
@@ -267,6 +290,15 @@ func ProviderConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Pr
 		config.Credentials = v.(string)
 	}
 
+
+	if v, ok := d.GetOk("external_credentials"); ok {
+		externalCredentials, err := transport_tpg.ExpandExternalCredentials(v)
+		if err != nil {
+			return nil, diag.FromErr(err)
+		}
+		config.ExternalCredentials = externalCredentials
+	}
+
 	// only check environment variables if neither value was set in config- this
 	// means config beats env var in all cases.
 	if config.AccessToken == "" && config.Credentials == "" {
```

Comment:
Wouldn't we want to exclude checking for the other credentials here if we've already found External Credentials?

Otherwise, have you tested what happens if we have both these environment variables and external credentials?

Reply:
https://github.com/GoogleCloudPlatform/magic-modules/pull/12714/commits/1470e65208d45ce850491ba153e8f7235d9a6359 - will not check the other auth fields if `external_credentials` is set

We also prevent the use of environment variables if `external_credentials == nil` since configuration takes precedence over environment variables.